src/Display/Gallery/Gallery.tsx

```diff
@@ -15,143 +15,112 @@ import {
-      visible: props.initialVisibility,
-      currentIndex: 0,
-      imageLeft: 0,
-    };
-    this.sliderRef = React.createRef();
-  }
-
-  closeModal = () => {
-    this.setState({
-      visible: false,
-    });
+const Gallery = ({
+  initialVisibility,
+  children,
+  imagesDisplayed = defaultImagesDisplayed,
+}: Props) => {
+  const sliderRef = React.useRef<HTMLDivElement>();
+
+  const [visible, setVisible] = React.useState<boolean>(initialVisibility);
+  const [currentIndex, setCurrentIndex] = React.useState<number>(0);
+  const [imageLeft, setImageLeft] = React.useState<number>(0);
+
+  const closeModal = () => setVisible(false);
+
+  const handleClick = (index: number) => {
+    setVisible(true);
+    setCurrentIndex(index);
   };
 
-  handleClick = (index: number) => {
-    this.setState({
-      visible: true,
-      currentIndex: index,
-    });
-  };
-
-  handleClickThumbs = (index: number) => {
-    this.setState({
-      currentIndex: index,
-    });
-  };
-
-  getCurrentIndex = (index: number) => {
-    this.setState({
-      currentIndex: index - 1,
-    });
-  };
-
-  componentDidMount() {
-    const { children, imagesDisplayed = defaultImagesDisplayed } = this.props;
-    if (React.Children.count(children) > imagesDisplayed) {
-      this.setState({
-        imageLeft: React.Children.count(children) - imagesDisplayed,
-      });
-    }
-    if (this.state.visible) {
-      this.sliderRef.current.focus();
-    }
-  }
-
-  componentDidUpdate(prevProps: Props, prevState: State) {
-    if (!prevState.visible && this.state.visible && this.sliderRef.current) {
-      this.sliderRef.current.focus();
-    }
-  }
-
-  render() {
-    const { children, imagesDisplayed = defaultImagesDisplayed } = this.props;
-    const { visible, currentIndex, imageLeft } = this.state;
-    return (
-      <GalleryContainer className="aries-gallery">
-        <GalleryItemWrapper className="gallery-wrapper">
-          {React.Children.toArray(children)
-            .slice(0, imagesDisplayed)
-            .map((data: React.ReactElement<React.HTMLProps<'img'>>, index) => (
-              <GalleryItem
-                className="gallery-item"
-                key={`${data.props.src}_${index}`}
-                imageLeft={imageLeft}
-                onClick={() => this.handleClick(index)}
-              >
-                <img src={data.props.src} alt={index.toString(10)} />
-              </GalleryItem>
-            ))}
-        </GalleryItemWrapper>
-        <Modal
-          isVisible={visible}
-          onClose={this.closeModal}
-          hideContentArea
-          removeAnimation
+  const handleClickThumbs = (index: number) => setCurrentIndex(index);
+
+  const getCurrentIndex = (index: number) => setCurrentIndex(index - 1);
+
+  React.useEffect(function componentDidMount() {
+    if (React.Children.count(children) > imagesDisplayed)
+      setImageLeft(React.Children.count(children) - imagesDisplayed);
+    if (visible) sliderRef.current.focus();
```

Comment:
The next useEffect adds the focus to the slider, any reason to add this here?

Reply:
No you're right, it can be removed. Thanks!

Reply:
Thanks!